pkg/apb/secrets.go

```diff
@@ -0,0 +1,159 @@
+package apb
+
+import (
+	"sync"
+
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	logging "github.com/op/go-logging"
+	"github.com/openshift/ansible-service-broker/pkg/clients"
+)
+
+// SecretsConfig - Entry for a secret config block in broker config
+type SecretsConfig struct {
+	Title   string `yaml:"title"`
+	ApbName string `yaml:"apb_name"`
+	Secret  string `yaml:"secret"`
+}
+
+// Validate - Ensures that the secrets config is valid (ie, all strings are non-empty
+func (c SecretsConfig) Validate() bool {
+	for _, str := range []string{c.Title, c.ApbName, c.Secret} {
+		if str == "" {
+			return false
+		}
+	}
+	return true
+}
+
+// AssociationRule - A rule to associate apbs with a secrets
+type AssociationRule struct {
+	apbName string
+	secret  string
+}
+
+type secretsCache struct {
+	mapping map[string][]string
+	rwSync  sync.RWMutex
+	rules   []AssociationRule
+	config  []SecretsConfig
+	log     *logging.Logger
+}
+
+var secrets secretsCache
+
+// GetSecrets - Returns a list of secrets to be attached to a specified spec
+func GetSecrets(spec *Spec) []string {
+	secrets.rwSync.RLock()
+	defer secrets.rwSync.RUnlock()
+	return secrets.mapping[spec.FQName]
+}
+
+// AddSecrets - Uses the AssociationRules generated from config to link specs to secrets and add them to the global secrets cache
+func AddSecrets(specs []*Spec) {
+	for _, spec := range specs {
+		AddSecretsFor(spec)
+	}
+}
+
+// AddSecretsFor - Uses AssociationRules for a given spec to link the spec to secrets and add them to the global secrets cache
+func AddSecretsFor(spec *Spec) {
+	secrets.rwSync.Lock()
+	defer secrets.rwSync.Unlock()
+
+	for _, rule := range secrets.rules {
+		if match(spec, rule) {
+			addSecret(spec, rule)
+		}
+	}
+
+}
+
+func addSecret(spec *Spec, rule AssociationRule) {
+	secrets.mapping[spec.FQName] = append(secrets.mapping[spec.FQName], rule.secret)
+}
+
+func match(spec *Spec, rule AssociationRule) bool {
+	return spec.FQName == rule.apbName
+}
```

Comment:
Yeah, I separated a lot of these functions out pre-emptively because I anticipate that they will become significantly more complex with the second phase of these changes.